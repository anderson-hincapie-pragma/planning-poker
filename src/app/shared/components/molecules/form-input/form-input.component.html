<div class="m-form-input">
    <label class="m-form-input__label">{{labelText}}</label>
    <app-input class="m-form-input__input" [type]="inputType" [placeholder]="inputPlaceholder" [control]="inputControl"></app-input>
</div>