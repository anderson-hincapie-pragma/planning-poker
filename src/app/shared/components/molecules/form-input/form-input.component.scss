:host {
    display: contents;
}

.m-form-input {
    width: 100%;
    height: fit-content;
    display: flex;
    flex-direction: column;
    gap: 8px;

    .m-form-input__label {
        font-family: 'Poppins';
        font-style: normal;
        font-weight: 700;
        font-size: 14px;
        line-height: 140%;
        color: #FFFFFF;
    }
}