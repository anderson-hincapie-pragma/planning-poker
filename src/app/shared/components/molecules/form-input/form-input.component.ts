import { Component, Input } from '@angular/core';
import { FormControl } from '@angular/forms';

@Component({
  selector: 'app-form-input',
  templateUrl: './form-input.component.html',
  styleUrls: ['./form-input.component.scss']
})
export class FormInputComponent {

  @Input()
  public labelText: string = 'Label';

  @Input()
  public inputType: 'email' | 'number' | 'search' | 'text' = 'text';

  @Input()
  public inputPlaceholder: string = '';

  @Input()
  public inputControl: FormControl = new FormControl();

}
