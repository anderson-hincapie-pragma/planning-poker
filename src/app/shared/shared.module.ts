import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { ActionButtonComponent } from './components/atoms/action-button/action-button.component';
import { InputComponent } from './components/atoms/input/input.component';
import { FormsModule, ReactiveFormsModule } from '@angular/forms';
import { FormInputComponent } from './components/molecules/form-input/form-input.component';
import { InputCardComponent } from './components/organisms/input-card/input-card.component';
import { CreateGameTemplateComponent } from './components/templates/create-game-template/create-game-template.component';



@NgModule({
  declarations: [
    ActionButtonComponent,
    InputComponent,
    FormInputComponent,
    InputCardComponent,
    CreateGameTemplateComponent
  ],
  imports: [
    CommonModule,
    ReactiveFormsModule
  ],
  exports: [
    ActionButtonComponent,
    InputComponent,
    FormInputComponent,
    InputCardComponent,
    CreateGameTemplateComponent
  ]
})
export class SharedModule { }
