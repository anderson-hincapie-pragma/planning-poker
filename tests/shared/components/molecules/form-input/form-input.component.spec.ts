import { ComponentFixture, TestBed } from '@angular/core/testing';

import { FormInputComponent } from '@shared/components/molecules/form-input/form-input.component';
import { SharedModule } from '@shared/shared.module';

describe('FormInputComponent', () => {
  let component: FormInputComponent;
  let fixture: ComponentFixture<FormInputComponent>;
  let compiled: HTMLElement;

  beforeEach(() => {
    TestBed.configureTestingModule({
      imports: [SharedModule],
      declarations: [FormInputComponent]
    });
    fixture = TestBed.createComponent(FormInputComponent);
    compiled = fixture.nativeElement;
    component = fixture.componentInstance;
    fixture.detectChanges();
  });

  it('should create', () => {
    expect(component).toBeTruthy();
  });

  it('shoud match snapshot', () => {
    expect(compiled).toMatchSnapshot();
  });

});
