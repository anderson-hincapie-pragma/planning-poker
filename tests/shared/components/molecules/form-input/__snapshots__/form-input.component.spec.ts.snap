// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormInputComponent shoud match snapshot 1`] = `
<div
  id="root1"
>
  <div
    class="m-form-input"
  >
    <label
      class="m-form-input__label"
    >
      Label
    </label>
    <app-input
      class="m-form-input__input"
    >
      <input
        class="a-input ng-untouched ng-pristine ng-valid"
        placeholder=""
        type="text"
      />
    </app-input>
  </div>
</div>
`;
